pkg/validation/validate_component.go

```diff
@@ -47,75 +50,108 @@ func NewComponentValidator(c []*bundle.ComponentPackage, set *bundle.ComponentSe
 }
 
 // Validate validates components and components sets, providing as many errors as it can.
-func (v *ComponentValidator) Validate() []error {
-	var errs []error
+func (v *ComponentValidator) Validate() field.ErrorList {
+	errs := field.ErrorList{}
 	errs = append(errs, v.validateComponentsSet()...)
 	errs = append(errs, v.validateComponentPackages()...)
 	errs = append(errs, v.validateObjects()...)
 	return errs
 }
 
-func (v *ComponentValidator) validateComponentPackages() []error {
-	var errs []error
+func cPath(ref bundle.ComponentReference) *field.Path {
+	return field.NewPath(fmt.Sprintf("Component{%s, %s}", ref.ComponentName, ref.Version))
+}
+
+func (v *ComponentValidator) validateComponentPackages() field.ErrorList {
+	cpathIdx := func(idx int) *field.Path {
+		return field.NewPath("Component").Index(idx)
+	}
+
+	errs := field.ErrorList{}
 	objCollect := make(map[bundle.ComponentReference]bool)
-	for _, ca := range v.components {
+	for i, ca := range v.components {
 		n := ca.Spec.ComponentName
-		if err := validateName(n); err != nil {
-			errs = append(errs, fmt.Errorf("the component name %q was invalid config %v", n, ca))
+		if n == "" {
+			errs = append(errs, field.Required(cpathIdx(i).Child("Spec", "ComponentName"), "componentName is required"))
+		}
+		ver := ca.Spec.Version
+		if ver == "" {
+			errs = append(errs, field.Required(cpathIdx(i).Child("Spec", "Version"), "version is required"))
+		}
+		if n == "" || ver == "" {
+			// Other validation relies on components having a unique name+version pair.
```

Comment:
s/Other/Subsequent
Is uniqueness of (n, ver) across the components a requirement of below validation? If not could just say "non-empty" in this comment.

Reply:
uniqueness is a requirement

Reply:
Were we testing for uniqueness? Maybe I missed that?